ui/src/datasets.tsx

```diff
@@ -819,44 +819,58 @@ function ExportDialog(
 
         window.location.href = result.redirectURL;
       } else {
-        // TODO(tjennison): Use plugins to handle different types of output.
-        const annotations: { cohortId: string; url: string }[] = [];
-        const data: { occurrenceId: string; url: string }[] = [];
+        const artifactLists: {
+          section: string;
+          name: string;
+          url: string;
+        }[][] = [];
 
         for (const key in result.outputs) {
           const parts = key.split(":");
-          if (parts.length !== 2) {
+          if (parts.length > 2) {
             throw new Error(`Invalid output key ${key}.`);
           }
 
-          if (parts[0] === "cohort") {
-            annotations.push({ cohortId: parts[1], url: result.outputs[key] });
-          } else if (parts[0] === "entity") {
-            data.push({ occurrenceId: parts[1], url: result.outputs[key] });
+          const artifact = {
+            section: parts.length > 1 ? parts[0] : "",
+            name: parts.length > 1 ? parts[1] : parts[0],
+            url: result.outputs[key],
+          };
+          const list = artifactLists.find(
+            (list) => list[0].section === artifact.section
+          );
+          if (list) {
+            list.push(artifact);
           } else {
-            throw new Error(`Unknown output type ${parts[0]}`);
+            artifactLists.push([artifact]);
           }
         }
 
+        artifactLists.sort((a, b) => a[0].section.localeCompare(b[0].section));
+
+        // TODO(tjennison): Remove fallback for looking up cohort ids once the
+        // backend plugin returns them.
```

Comment:
Does this refer to the backend service or a plugin in the UI?

Reply:
The backend service. This is now displaying whatever the export plugin returns so the plugin should be updated to display what we want to see here rather than relying on the UI to do translation. I just added a step that checks if the name is a valid cohort id, in which case I look up the cohort name and display that instead. That's a bit of a hack though  so it would be better to remove it.

Reply:
I'm still optionally processing the `:`s to generate category headings, so probably something like:
```
Data:person -->  <person_file_url>
Data:condition_occurrence -->  <person_file_url>
Annotations:Women with diabetes -->  <annotation_file_url>
Annotations:Men with diabetes -->  <annotation_file_url>
```

I'll process that into:
```
Annotations
- Women with diabetes
- Men with diabetes
Data
- person
- condition_occurrence
```

Where each item in the lists is a link to the corresponding URL.